src/main/java/com/parasoft/environmentmanager/jenkins/ExecuteJobBuilder.java

```diff
@@ -174,6 +194,11 @@ public void logMessage(String message) {
 						reportDir.mkdirs();
 						FilePath reportXmlFile = new FilePath(reportDir, "report.xml");
 						reportXmlFile.copyFrom(jobs.download("testreport/" + reportIds.getLong(i) + "/report.xml"));
+						
+						if(publish) {
+						    result = publishReport(reportXmlFile, listener.getLogger()) && result;
```

Comment:
This result boolean is used to fail the build step if any of the functional tests had failures.  This line will incorrectly change that failure state to success if the report successfully uploads to DTP.

Reply:
that is why i have && result at the end. I have been testing with uploading failure reports, the job always has failure status when finished.

Reply:
Okay, that makes sense.  I didn't notice that at the end of the line.  I was focusing on the assignment operator because my convention on line 173 was to use result &= ...